src/classifyarray.jl

```diff
@@ -0,0 +1,13 @@
+function w2cp(vec) 
+    v = cumsum(vec)
+    v ./= v[end]
+end
+
+function calcBoundaries(array, cumulativeProportions, classifyMask = nothing)
+    maskedarray = isnothing(classifyMask) ? array : mask!(copy(array))
+    nCells = count(isfinite, maskedarray)
+    boundaryIndexes = ceil(Int, cumulativeProportions * nCells)
+    boundaryValues = sort(vec(maskedarray))[boundaryIndexes]
```

Comment:
there are two full copies in this function - it must be possible to avoid at least this one